Contents/Code/__init__.py

```diff
@@ -126,140 +213,262 @@ def Update(self, metadata, media, lang, force, movie):
+        studio = try_get(studio, 0, False)
+        if not studio:
+            studio = HttpReq('api/v3/Series/%s/Cast?roleType=Staff&roleDetails=Work' % aid) # http://127.0.0.1:8111/api/v3/Series/24/Cast?roleType=Staff&roleDetails=Work
+            studio = try_get(studio, 0, False)
+        if studio:
+            Log('Studio: %s', studio['Staff']['Name'])
+            metadata.studio = studio['Staff']['Name']
 
         if not movie:
-            for ep in series['eps']:
-                if ep['eptype'] not in ["Episode", "Special", "Credits", "Trailer"]:
-                    continue
-
-                if ep['eptype'] == "Episode": season = 1
-                elif ep['eptype'] == "Special": season = 0
-                elif ep['eptype'] == "Credits": season = -1
-                elif ep['eptype'] == "Trailer": season = -2
-                if not Prefs['SingleSeasonOrdering']:
-                    try:
-                        season = int(ep['season'].split('x')[0])
-                        if season <= 0 and ep['eptype'] == 'Episode': season = 1
-                        elif season > 0 and ep['eptype'] == 'Special': season = 0
-                    except:
-                        pass
-
-                episodeObj = metadata.seasons[season].episodes[ep['epnumber']]
-                episodeObj.title = ep['name']
-                if (ep['summary'] != "Episode Overview not Available"): 
-                    episodeObj.summary = summary_sanitizer(ep['summary'])
-                Log("" + str(ep['epnumber']) + ": " + ep['summary'])
-
-                airdate = try_get(ep, 'air', '1/01/0001 12:00:00 AM')
-
-                if airdate != '1/01/0001 12:00:00 AM' and airdate != '0001-01-01':
-                    episodeObj.originally_available_at = datetime.strptime(airdate, "%Y-%m-%d").date()
-
-                if len(ep['art']['thumb']) and Prefs['customThumbs']:
-                    self.metadata_add(episodeObj.thumbs, ep['art']['thumb'])
-
-            links = HttpReq("api/links/serie?id=%s" % aid)
+            # Get episode list using series ID
+            episodes = HttpReq('api/v3/Series/%s/Episode' % aid) # http://127.0.0.1:8111/api/v3/Series/212/Episode
+
+            for episode in episodes:
+                # Get episode data
+                ep_id = episode['IDs']['ID']
+                ep_data = {}
+                ep_data['anidb'] = HttpReq('api/v3/Episode/%s/AniDB' % ep_id)
+                ep_data['tvdb'] = HttpReq('api/v3/Episode/%s/TvDB' % ep_id)
+
+                ep_type = ep_data['anidb']['Type']
+
+                # Get season number
+                season = 0
+                episode_number = None
+                if ep_type == 'Normal': season = 1
+                elif ep_type == 'Special': season = 0
+                elif ep_type == 'ThemeSong': season = -1
+                elif ep_type == 'Trailer': season = -2
+                elif ep_type == 'Parody': season = -3
+                elif ep_type == 'Unknown': season = -4
```

Comment:
These need to be rethought - plex doesn't like anything below -1 and will bug out these days.

Reply:
Is this an issue with certain Plex clients or something? Negative seasons work correctly on Desktop (Windows)/Android/Roku/iOS so I see no evidence on my end of this actually causing problems. The alternative to avoid any episode overlaps would be to just assign huge season numbers to these instead (in either the hundreds or thousands) but then regular seasons of a show would end up sandwiched between specials and 'other' seasons which is somewhat awkward.

Reply:
I've just seen this having issues and be discussion last year: https://discord.com/channels/96234011612958720/268484849419943936/847683155502563389 Plex may have fixed it up, though I honestly don't trust them to keep it in a working state.

Reply:
It seems like that issue has long been fixed and from what I can gather in that discord discussion large season numbers aren't desired either. The only other option would be lumping everything in season zero with different episode number offsets depending on the ep_type. It works perfectly fine though as is so I don't really see the point of doing so.

The way I see it, the chances of it breaking in a Plex update are likely the same as anything else in the agent breaking.

Reply:
Yeah, I wasn't really saying this **needs** to be changed now, just, moreso needs to be rethought in the long term, as it is a hacky way, I am honestly more on the land of just dropping support of showing NCED/NCOP, and stuff that wouldn't be a specific special (Like parody, etc) and just have everything supported in season -1.

Reply:
Sounds good, the only thing that is needed alongside specials is the 'Unknown' (Other) category. There are quite a few shows that use that for regular episodes and they are generally able to be tvdb/tmdb mapped too.

For everything else, it should be possible to populate [local extras](https://support.plex.tv/articles/local-files-for-trailers-and-extras/#toc-2) with the agent. I'll look into how well that works in the next couple weeks as a potential fallback.